sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_publisher_client_async.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
+
+from typing import Any
+
+from azure.core import AsyncPipelineClient
+from msrest import Deserializer, Serializer
+
+from .._models import CloudEvent, EventGridEvent
+from azure.core.pipeline.policies import AzureKeyCredentialPolicy
+from azure.core.credentials import AzureKeyCredential
+from .._generated.event_grid_publisher_client.aio._event_grid_publisher_client_async import EventGridPublisherClient as EventGridPublisherClientAsync
+from .. import _constants as constants
+
+
+class EventGridPublisherClient(object):
+    """Asynchronous EventGrid Python Publisher Client.
+
+    :param str topic_hostname: The topic endpoint to send the events to.
+    :param credential: The credential object used for authentication which implements SAS key authentication or SAS token authentication.
+    :type credential: Union[~azure.core.credentials.AzureKeyCredential, ~azure.core.credentials.TokenCredential]
+    """
+
+    pass
+    def __init__(
+        self,
+        topic_hostname,  # type: str
+        credential,  # type: Union[azure.core.credential.AzureKeyCredential, azure.core.credential.TokenCredential]
+        **kwargs  # type: Any
+    ):
+        # type: (str, Union[AzureKeyCredential, TokenCredential], Any) -> None
+        auth_policy = EventGridPublisherClient._get_authentication_policy(credential)
+        self._client = EventGridPublisherClientAsync(authentication_policy=auth_policy)
+        self._credential = credential
+        self._topic_hostname = topic_hostname
+
+
+    async def send(
+        self,
+        events,
+        **kwargs
+    ):
+        # type: (Union[List[CloudEvent], List[EventGridEvent], List[CustomEvent], dict], Any) -> None
+        """Sends event data to topic hostname specified during client initialization.
+
+        :param  events: A list of CloudEvent/EventGridEvent/CustomEvent to be sent. If a dict is sent, it will be interpreted as a CloudEvent.
+        :type events: Union[List[models.CloudEvent], List[models.EventGridEvent], List[models.CustomEvent], dict]
+        :rtype: None
+         """
+
+        if isinstance(events[0], CloudEvent):
+            self._client.publish_cloud_event_events(self._topic_hostname, events)
+        elif isinstance(events[0], EventGridEvent):
+            self._client.publish_event_grid_events(self._topic_hostname, events)
+        else:
+            print("Event schema is not correct. Please send as list of CloudEvent or list of EventGridEvent.")
+    
+    @classmethod
+    def _get_authentication_policy(cls, credential):
```

Comment:
Why not static since you don't use cls?

Reply:
I was following the Python guidelines here, which mention that we should avoid static methods.: https://azure.github.io/azure-sdk/python_introduction.html#method-signatures.

I changed it to an instance method, with the method signature as `def _get_authentication_policy(self)`. This way, I can get the credential inside the method by calling self._credential, instead of passing it in. So from line 37-40, I now have:
```self._credential = credential
        self._topic_hostname = topic_hostname
        auth_policy = self._get_authentication_policy()
        self._client = EventGridPublisherClientImpl(authentication_policy=auth_policy, **kwargs)```

Reply:
"Do NOT use static methods. Prefer module level functions instead". So let's make it module level then :)